### PR TITLE
Get semantic-ui theme on react 17 and dts-cli

### PR DESCRIPTION
**Pull request by heath-freenome**
### Reasons for making this change

- Updated the `package*.json` to update to react 17 and switch from `tsdx` to `dts-cli`
- Ran `cs-format` over `semantic-ui` after upgrading to `dts-cli` that uses a newer version of `prettier`
- Updated the tests to mock the `Ref` component from `@fluentui/react-component-ref` because it causes tests to fail
- Added the same additional Array tests to other themes that were added in https://github.com/rjsf-team/react-jsonschema-form/pull/2576
- Updated the `5.x migration guide` and the `CHANGELOG.md`

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [ ] **I'm adding or updating code**
  - [ ] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/semantic-ui/test/Form.test.js [hunk 1]
```diff
@@ -4,6 +4,12 @@ import renderer from "react-test-renderer";
 
 import Form from "../src/index";
 
+/** Mock the `react-component-ref` component used by semantic-ui to simply render the children, otherwise tests fail */
+jest.mock("@fluentui/react-component-ref", () => ({
```
Comment (jacqueswho) [inline packages/semantic-ui/test/Form.test.js:8]: Should this be semantic-ui

File: packages/semantic-ui/test/Array.test.js [hunk 2]
```diff
@@ -4,6 +4,12 @@ import renderer from "react-test-renderer";
 
 import Form from "../src/index";
 
+/** Mock the `react-component-ref` component used by semantic-ui to simply render the children, otherwise tests fail */
+jest.mock("@fluentui/react-component-ref", () => ({
```
Comment (jacqueswho) [inline packages/semantic-ui/test/Array.test.js:8]: Should this be semantic-ui

File: packages/semantic-ui/test/Form.test.js [hunk 3]
```diff
@@ -4,6 +4,12 @@ import renderer from "react-test-renderer";
...
+  ...jest.requireActual("@fluentui/react-component-ref"),
```
Comment (jacqueswho) [inline packages/semantic-ui/test/Form.test.js:9]: Should this be semantic-ui

File: packages/semantic-ui/test/Object.test.js [hunk 4]
```diff
@@ -4,6 +4,12 @@ import renderer from "react-test-renderer";
 
 import Form from "../src/index";
 
+/** Mock the `react-component-ref` component used by semantic-ui to simply render the children, otherwise tests fail */
+jest.mock("@fluentui/react-component-ref", () => ({
```
Comment (jacqueswho) [inline packages/semantic-ui/test/Object.test.js:8]: Should this be semantic-ui

**Review by jacqueswho — APPROVED** [review]

File: packages/semantic-ui/test/Form.test.js (hunk 1 above)
Comment (heath-freenome) [reply packages/semantic-ui/test/Form.test.js:8]: @jacqueswho Actually, it needs to be `@fluentui/react-component-ref`... Looking at the [semantic-ui-react source code](https://github.com/Semantic-Org/Semantic-UI-React/blob/master/src/elements/Button/Button.js#L1) you can see that is what needs to be mocked

**Pull request merged**